pkg/controller/garbagecollector/garbagecollector_test.go

```diff
@@ -920,6 +928,102 @@ func TestGarbageCollectorSync(t *testing.T) {
 	if err != nil {
 		t.Fatalf("Expected garbagecollector.Sync to still be running but it is blocked: %v", err)
 	}
+
+	// Add fake controller which is never synced to simulate the not synced informer
+	fc := fakeController{}
+	gc.dependencyGraphBuilder.monitors[schema.GroupVersionResource{
+		Version:  "v1",
+		Resource: "secrets",
+	}] = &monitor{controller: &fc}
+
+	fakeDiscoveryClient.setPreferredResources(unsyncableServerResources)
+	fakeDiscoveryClient.setError(nil)
+
+	err = expectSyncNotBlocked(fakeDiscoveryClient, &gc.workerLock)
+	if err != nil {
+		t.Fatalf("Expected garbagecollector.Sync to still be running but it is blocked: %v", err)
+	}
+
+	fc.synced = true
```

Comment:
if we do end up needing the stub monitor, does this need to be made threadsafe? I'm not sure whether there are background goroutines checking this

Reply:
It should be threadsafe.